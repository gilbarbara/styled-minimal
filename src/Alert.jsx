import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

const styles = (props: Object): string => {
  const { size } = props;
  const { borderRadius, maxWidth, padding } = themeGet(props, 'alert');

  return css`
    ${baseStyles.variant};
    border-radius: ${px(borderRadius)};
    font-size: ${baseStyles.fontSize};
    line-height: ${baseStyles.lineHeight};
    max-width: ${px(maxWidth)};
    padding: ${px(padding[size][0])} ${px(padding[size][1])};
    width: 100%;

    a {
      ${baseStyles.color};
    }
  `;
};

const Alert = styled(Box).attrs({
  role: 'alert',
})`
  ${styles};
`;

Alert.displayName = 'Alert';

Alert.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  /** button size */
  size: sizesAllPropTypes,
  /** button variant */
  variant: variantPropTypes,
  ...Box.propTypes,
};

Alert.defaultProps = {
  outline: false,
  size: 'md',
  variant: 'primary',
};

export default Alert;
