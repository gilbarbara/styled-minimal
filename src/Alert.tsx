import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import {
  baseStyles,
  colorStyles,
  lineHeightStyles,
  sx,
  system,
  variantStyles,
} from './utils/system';

import { AlertProps } from './types';

export const StyledAlert = styled('div')<AlertProps>(
  props => {
    const { borderRadius, maxWidth, padding } = getTheme(props, 'alert');

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${px(borderRadius)};
      font-size: inherit;
      line-height: ${lineHeightStyles};
      max-width: ${px(maxWidth)};
      padding: ${padding};
      width: 100%;

      a {
        ${colorStyles};
      }
    `;
  },
  sx,
  system,
);

export const Alert = React.forwardRef<HTMLDivElement, AlertProps>((props, ref) => (
  <StyledAlert ref={ref} {...props} />
));

Alert.defaultProps = {
  dark: false,
  invert: false,
  role: 'alert',
  variant: 'primary',
};
