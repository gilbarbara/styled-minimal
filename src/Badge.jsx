import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

export const Badge = styled(Box)`
  ${(props: Object): string => {
    const {
      borderRadius,
      fontSize,
      fontWeight,
      padding,
    } = themeGet(props, 'badge');
    const fontSizeProp = themeGet(props, 'componentSizes', { key: 'size' });

    return css`
      ${baseStyles.variant};
      border-radius: ${px(borderRadius)};
      display: inline-flex;
      font-size: ${fontSize};
      ${fontSizeProp ? `font-size: ${fontSizeProp}` : ''};
      font-weight: ${fontWeight};
      line-height: 1;
      padding: ${px(padding[0])} ${px(padding[1])};
      vertical-align: baseline;
    `;
  }};
`;

Badge.displayName = 'Badge';

Badge.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  size: sizesAllPropTypes,
  variant: variantPropTypes,
  ...Box.propTypes,
};

Badge.defaultProps = {
  as: 'span',
};

export default Badge;
