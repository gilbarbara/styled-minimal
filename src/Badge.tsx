import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, sx, system, variantStyles } from './utils/system';

import { BadgeProps } from './types';

export const StyledBadge = styled('sup')<BadgeProps>(
  props => {
    const { dot } = props;
    const { borderRadius, fontSize, fontWeight, lineHeight, padding, size, sizeDot } = getTheme(
      props,
      'badge',
    );

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${dot ? '100%' : px(borderRadius)};
      display: inline-flex;
      font-size: ${dot ? 0 : px(fontSize)};
      font-weight: ${fontWeight};
      justify-content: center;
      line-height: ${dot ? 0 : px(lineHeight)};
      height: ${px(dot ? sizeDot : size)};
      padding: ${dot ? 0 : `0 ${px(padding)}`};
      min-width: ${px(dot ? sizeDot : size)};
    `;
  },
  sx,
  system,
);

export const Badge = React.forwardRef<HTMLSpanElement, BadgeProps>((props, ref) => (
  <StyledBadge ref={ref} {...props} />
));

Badge.defaultProps = {
  dark: false,
  dot: false,
  invert: false,
  variant: 'primary',
};
