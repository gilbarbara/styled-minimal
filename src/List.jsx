import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { ListStyles } from './utils/system';
import { sizeType } from './utils/types';

export const StyledList = styled.ul`
  ${ListStyles.base};

  > li {
    ${ListStyles.item}
  
    + li {
      ${ListStyles.itemSibling}
    }
  }
`;

const List = ({ children, ...props }) => (
  <StyledList {...props}>{children}</StyledList>
);

List.propTypes = {
  /** element type */
  as: PropTypes.oneOf(['ul', 'ol']),
  bordered: PropTypes.bool,
  children: PropTypes.node.isRequired,
  inline: PropTypes.bool,
  reversed: PropTypes.bool,
  size: sizeType,
  start: PropTypes.number,
  styleType: PropTypes.oneOfType([
    PropTypes.oneOf([
      'disc',
      'circle',
      'square',
      'decimal',
      'lower-alpha',
      'none',
    ]),
    PropTypes.string,
  ]),
  type: PropTypes.oneOf(['1', 'a', 'A', 'i', 'I']),
};

List.defaultProps = {
  as: 'ul',
  bordered: false,
  inline: false,
  size: 'md',
  styleType: 'none',
  type: '1',
};

export default List;
