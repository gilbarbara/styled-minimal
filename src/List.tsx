import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, sx, system } from './utils/system';

import { DLProps, OLProps, ULProps } from './types';

const stylesItem = (props: any) => {
  const { size = 'md' } = props;
  const { padding } = getTheme(props, 'list');

  return css`
    padding: ${px(padding[size])};
  `;
};

const stylesDD = (props: any) => {
  const { size = 'md' } = props;
  const { padding } = getTheme(props, 'list');

  return css`
    margin-left: ${px(padding[size])};
    padding: 0 ${px(padding[size])} ${px(padding[size])} 0;
  `;
};

const stylesSibling = (props: any) => {
  const { frame } = props;
  const { borderColor } = getTheme(props, 'list');

  if (!frame) {
    return undefined;
  }

  return css`
    border-top: 1px solid ${borderColor};
  `;
};

const stylesChildren = (props: any) => {
  const { as, frame, inline, styleType } = props;
  const { borderColor } = getTheme(props, 'list');
  const border = `1px solid ${borderColor}`;

  if (as === 'dl') {
    return css`
      dt {
        font-weight: 700;
        ${stylesItem}
        ${frame && inline && `border-bottom: ${border};`};
        ${frame && inline && `border-right: ${border};`};
        ${inline && `flex-basis: 30%;`};
      }

      dd {
        ${stylesDD};
        ${frame && inline && `border-bottom: ${border};`};
        ${inline && stylesItem};
        ${inline && `flex-basis: 70%;`};
        ${inline && `margin-left: 0;`};

        + dt {
          ${!inline && stylesSibling};
        }

        &:last-of-type {
          ${inline && `border-bottom: 0;`};
        }
      }

      dt:last-of-type {
        ${inline && `border-bottom: 0;`};
      }
    `;
  }

  return css`
    > li {
      ${frame && (as === 'ol' || styleType !== 'none' || inline) && `border-left: ${border};`};
      ${inline && 'flex: 1'};
      line-height: 1.2;
      ${stylesItem}

      + li {
        ${stylesSibling}
      }
    }
  `;
};

export const StyledList = styled('ul')<any>(
  props => {
    const { as, frame, inline, styleType } = props;
    const { borderColor, borderRadius } = getTheme(props, 'list');
    const border = `1px solid ${borderColor}`;

    return css`
      ${baseStyles};
      ${frame && `border: ${border}`};
      ${frame && `border-radius: ${px(borderRadius)}`};
      display: flex;
      flex-direction: ${inline ? 'row' : 'column'};
      flex-wrap: ${as === 'dl' && inline ? 'wrap' : 'nowrap'};
      margin: 0;
      ${as === 'ul' && styleType === 'none' ? 'padding: 0;' : ''};
      ${as === 'ul' ? `list-style-type: ${styleType};` : ''};

      ${stylesChildren};
    `;
  },
  sx,
  system,
);

export const DL = React.forwardRef<HTMLDListElement, DLProps>((props, ref) => (
  <StyledList as="dl" ref={ref} {...props} />
));

DL.defaultProps = {
  frame: true,
  inline: false,
  size: 'md',
};

export const OL = React.forwardRef<HTMLOListElement, OLProps>((props, ref) => (
  <StyledList as="ol" ref={ref} {...props} />
));

OL.defaultProps = {
  frame: false,
  inline: false,
  size: 'md',
};

export const UL = React.forwardRef<HTMLUListElement, ULProps>((props, ref) => (
  <StyledList as="ul" ref={ref} {...props} />
));

UL.defaultProps = {
  frame: true,
  inline: false,
  size: 'md',
  styleType: 'none',
};
