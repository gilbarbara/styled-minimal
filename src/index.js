export { default as Alert } from './Alert';
export { default as Badge } from './Badge';
export { default as Box } from './Box';
export { default as Button } from './Button';
export { default as Code } from './Code';
export { default as Container } from './Container';
export { default as Embed } from './Embed';
export { default as Flex } from './Flex';
export { default as Fieldset } from './Fieldset';
export { default as Form } from './Form';
export { default as FormGroup } from './FormGroup';
export { default as Group } from './Group';
export { default as Heading } from './Heading';
export { default as Image } from './Image';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Legend } from './Legend';
export { default as Link } from './Link';
export { default as List } from './List';
export { default as Paragraph } from './Paragraph';
export { default as Screen } from './Screen';
export { default as Select } from './Select';
export { default as Switch } from './Switch';
export { default as Table } from './Table';
export { default as Text } from './Text';
export { default as Textarea } from './Textarea';
