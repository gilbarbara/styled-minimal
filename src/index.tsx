export { Alert } from './Alert';
export { Badge } from './Badge';
export { Box } from './Box';
export { Button } from './Button';
export { ButtonGroup } from './ButtonGroup';
export { Code } from './Code';
export { CodeBlock } from './CodeBlock';
export { Container } from './Container';
export { Embed } from './Embed';
export { Flex } from './Flex';
export { Fieldset } from './Fieldset';
export { Form } from './Form';
export { FormGroup } from './FormGroup';
export { Grid } from './Grid';
export { Group } from './Group';
export { Heading } from './Heading';
export { Image } from './Image';
export { Checkbox, Input, Radio } from './Input';
export { Label } from './Label';
export { Legend } from './Legend';
export { Link } from './Link';
export { DL, OL, UL } from './List';
export { Loader } from './Loader';
export { Paragraph } from './Paragraph';
export { RadioGroup } from './RadioGroup';
export { Select } from './Select';
export { Switch } from './Switch';
export { Table } from './Table';
export { Text } from './Text';
export { Textarea } from './Textarea';

export { default as system } from '@styled-system/css';
export { mergeTheme as getTheme, px, responsive } from './utils/helpers';
