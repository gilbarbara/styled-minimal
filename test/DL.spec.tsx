import * as React from 'react';
import { render } from '@testing-library/react';

import { DL } from '../src/List';

describe('DL', () => {
  const { container } = render(
    <DL>
      <dt>Beast of Bodmin</dt>
      <dd>
        Far far away, behind the word mountains, far from the countries Vokalia and Consonantia,
        there live the blind texts.
      </dd>
      <dt>Morgawr</dt>
      <dd>A sea serpent.</dd>
      <dt>Owlman</dt>
      <dd>A giant owl-like creature.</dd>
    </DL>,
  );

  it('should render properly', () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
