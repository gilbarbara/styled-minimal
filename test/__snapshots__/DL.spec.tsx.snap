// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DL should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  border: 1px solid #ccc;
  border-radius: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin: 0;
}

.c0 dt {
  font-weight: 700;
  padding: 12px;
}

.c0 dd {
  margin-left: 12px;
  padding: 0 12px 12px 0;
}

.c0 dd + dt {
  border-top: 1px solid #ccc;
}

<dl
  class="c0"
>
  <dt>
    Beast of Bodmin
  </dt>
  <dd>
    Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts.
  </dd>
  <dt>
    Morgawr
  </dt>
  <dd>
    A sea serpent.
  </dd>
  <dt>
    Owlman
  </dt>
  <dd>
    A giant owl-like creature.
  </dd>
</dl>
`;
